packages/apps/dashboard/ui-2024/src/features/Leaderboard/hooks/useDataGrid.tsx

```diff
@@ -0,0 +1,194 @@
+        field: 'role',
+        sortable: false,
+        flex: isMobile ? 0.8 : 1.5,
+        minWidth: isMobile ? 100 : 240,
+        headerClassName: isMobile
+          ? 'home-page-table-header pinned-column--header'
+          : 'home-page-table-header',
+        cellClassName: isMobile ? 'pinned-column--cell' : '',
+        renderHeader: () => (
+          <Typography component="span" variant="body3">
+            Role
+          </Typography>
+        ),
+        renderCell: (params: GridRenderCellParams) => (
+          <RoleCell role={params.value} />
+        ),
+      },
+      {
+        field: 'address',
+        sortable: false,
+        flex: 1,
+        minWidth: 150,
+        headerClassName: 'home-page-table-header',
+        renderHeader: () => (
+          <Box display="flex" gap="8px" alignItems="center">
+            <CustomTooltip title="Address of the role" arrow>
+              <HelpOutlineIcon
+                style={{
+                  color: colorPalette.sky.main,
+                }}
+              />
+            </CustomTooltip>
+            <Typography component="span" variant="body3">
+              Address
+            </Typography>
+          </Box>
+        ),
+        renderCell: (params: GridRenderCellParams) => (
+          <AddressCell chainId={params.row.chainId} address={params.value} />
+        ),
+      },
+      {
+        field: 'amountStaked',
+        flex: 1,
+        minWidth: 130,
+        headerClassName: 'home-page-table-header',
+        renderHeader: () => (
+          <Box display="flex" gap="8px" alignItems="center">
+            <CustomTooltip title="Amount of HMT staked" arrow>
+              <HelpOutlineIcon
+                style={{
+                  color: colorPalette.sky.main,
+                }}
+              />
+            </CustomTooltip>
+            <Typography component="span" variant="body3">
+              Stake
+            </Typography>
+          </Box>
+        ),
+        valueFormatter: (value: string) => {
+          return `${value} HMT`;
+        },
+        renderCell: (params: GridRenderCellParams) => (
+          <TextCell value={params.formattedValue} />
+        ),
+      },
+      {
+        field: 'chainId',
+        headerName: 'Network',
+        flex: isMobile ? 1 : 1.5,
+        sortable: false,
+        minWidth: isMobile ? 150 : 245,
+        headerClassName: 'home-page-table-header',
+        renderHeader: () => {
+          return (
+            <>
+              {isMobile ? (
+                <Typography component="span" variant="body3">
+                  Network
+                </Typography>
+              ) : (
+                <SelectNetwork />
+              )}
+            </>
+          );
+        },
+        renderCell: (params: GridRenderCellParams) => (
+          <ChainCell chainId={params.value} />
+        ),
+      },
+      {
+        field: 'reputation',
+        headerName: 'Reputation Score',
+        sortable: false,
```

Comment:
Reputation Score should remain sortable

Reply:
I disabled sorting because this feature is "coming soon"

![Zrzut ekranu 2024-11-21 o 10 48 06](https://github.com/user-attachments/assets/5d32ddf3-1d98-4353-a304-69330ec0985d)

Reply:
That makes sense, we can enable it when we start displaying some values 👍🏻